src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java

```diff
@@ -239,463 +245,66 @@ private Object createObject(String className) throws CheckstyleException {
-        NAME_TO_FULL_MODULE_NAME.put("FileContentsHolder",
-                BASE_PACKAGE + ".checks.FileContentsHolder");
-        NAME_TO_FULL_MODULE_NAME.put("FinalParametersCheck",
-                BASE_PACKAGE + ".checks.FinalParametersCheck");
-        NAME_TO_FULL_MODULE_NAME.put("NewlineAtEndOfFileCheck",
-                BASE_PACKAGE + ".checks.NewlineAtEndOfFileCheck");
-        NAME_TO_FULL_MODULE_NAME.put("OuterTypeFilenameCheck",
-                BASE_PACKAGE + ".checks.OuterTypeFilenameCheck");
-        NAME_TO_FULL_MODULE_NAME.put("SuppressWarningsHolder",
-                BASE_PACKAGE + ".checks.SuppressWarningsHolder");
-        NAME_TO_FULL_MODULE_NAME.put("TodoCommentCheck",
-                BASE_PACKAGE + ".checks.TodoCommentCheck");
-        NAME_TO_FULL_MODULE_NAME.put("TrailingCommentCheck",
-                BASE_PACKAGE + ".checks.TrailingCommentCheck");
-        NAME_TO_FULL_MODULE_NAME.put("TranslationCheck",
-                BASE_PACKAGE + ".checks.TranslationCheck");
-        NAME_TO_FULL_MODULE_NAME.put("UncommentedMainCheck",
-                BASE_PACKAGE + ".checks.UncommentedMainCheck");
-        NAME_TO_FULL_MODULE_NAME.put("UniquePropertiesCheck",
-                BASE_PACKAGE + ".checks.UniquePropertiesCheck");
-        NAME_TO_FULL_MODULE_NAME.put("UpperEllCheck",
-                BASE_PACKAGE + ".checks.UpperEllCheck");
+    private static Map<String, String> generateNameToFullModuleNameMap(ClassLoader loader) {
+        Map<String, String> returnValue;
+        try {
+            final Reflections reflections = new Reflections(new ConfigurationBuilder()
+                    .setScanners(new SubTypesScanner(false))
+                    .setUrls(ClasspathHelper.forPackage(BASE_PACKAGE))
+                    .filterInputsBy(new FilterBuilder()
+                            .include(FilterBuilder.prefix(BASE_PACKAGE))));
+            final Set<String> packageNames = PackageNamesLoader.getPackageNames(loader);
+            returnValue = reflections.getSubTypesOf(Object.class).stream()
+                    .filter(clazz -> packageNames.contains(clazz.getPackage().getName() + "."))
+                    .filter(PackageObjectFactory::shouldAddToMap)
+                    .collect(Collectors.toMap(Class::getSimpleName, Class::getCanonicalName));
+        } catch (CheckstyleException ignore) {
+            returnValue = new HashMap<>();
+        }
+        return returnValue;
     }
 
     /**
-     * Fill short-to-full module names map with modules from filefilters package.
+     * Whether the class should be added to the map.
+     * @param clazz the class to check
+     * @return true if the class should be added to the map
      */
-    private static void fillModulesFromFilefiltersPackage() {
-        NAME_TO_FULL_MODULE_NAME.put("BeforeExecutionExclusionFileFilter",
-                BASE_PACKAGE + ".filefilters.BeforeExecutionExclusionFileFilter");
+    private static boolean shouldAddToMap(Class<?> clazz) {
```

Comment:
@romani @Luolc 
Personally, I don't like this approach. We add so many conditions and new logic just for one thing. What are the benefits? We do not introduce new checks very frequently, so we can simply add the new checks into the map.